### PR TITLE
Add troubleshooting test email delivery (postfix)

### PR DESCRIPTION
**Pull request by asteflova**
#### What changes are you introducing?

Adding a troubleshooting bullet point regarding test emails not being delivered.

#### Why are you introducing these changes? (Explanation, links to references, issues, etc.)

https://issues.redhat.com/browse/SAT-27822

#### Anything else to add? (Considerations, potential downsides, alternative solutions you have explored, etc.)

N/A

#### Checklists

* [x] I am okay with my commits getting squashed when you merge this PR.
* [x] I am familiar with the [contributing](https://github.com/theforeman/foreman-documentation/blob/master/CONTRIBUTING.md) guidelines.

Please cherry-pick my commits into:

* [x] Foreman 3.12/Katello 4.14 (Satellite 6.16)
* [x] Foreman 3.11/Katello 4.13
* [x] Foreman 3.10/Katello 4.12
* [x] Foreman 3.9/Katello 4.11 (Satellite 6.15; orcharhino 6.8/6.9/6.10)
* [x] Foreman 3.8/Katello 4.10
* [x] Foreman 3.7/Katello 4.9 (Satellite 6.14)
* [x] Foreman 3.6/Katello 4.8
* [x] Foreman 3.5/Katello 4.7 (Satellite 6.13; orcharhino 6.6/6.7)
* We do not accept PRs for Foreman older than 3.5.

### CODE REVIEW

File: guides/common/modules/proc_testing-email-delivery.adoc [hunk 1]
```diff
@@ -17,3 +17,20 @@ Otherwise, you must perform the following diagnostic steps:
 .. Verify the user's email address.
 .. Verify {ProjectServer}'s email configuration.
 .. Examine firewall and mail server logs.
+.. {Project} uses the Postfix service for email delivery by default.
```
Comment (adamruzicka) [inline guides/common/modules/proc_testing-email-delivery.adoc:20]: With `-Wpedantic`:
`{Project}` doesn't use Postfix by default, `{Project}` uses `sendmail` by default. There are many implementations of `sendmail`, coming from various projects (postfix, opensmtpd, msmtp to name a few). EL* just happen to use Postfix as a sendmail provider, from the top of my head I'm not sure what's the default on debian-based distributions. In some of these implementations, sendmail just queues the mail to be delivered by a service, in some the sendmail command sends the email directly.

Comment (asteflova) [reply guides/common/modules/proc_testing-email-delivery.adoc]: Thanks! If I try to hedge a little, I can rephrase this to refer only to situations when Foreman really does use Postfix.

**Review by adamruzicka — APPROVED** [review]
Looks good

**Review by maximiliankolb — APPROVED** [review]

**Comment by asteflova** [comment]
Merged to "master" and cherry-picked:

d3d0c1e895..bf7f5cea04  3.12 -> 3.12
d66b8c24aa..96b9e714d2  3.11 -> 3.11
398118deca..bca3cd76c2  3.10 -> 3.10
530deeb3fd..27c1e9ac37  3.9 -> 3.9
3519b53092..ddad31ad43  3.8 -> 3.8
944e3b3106..9a09d8ebcc  3.7 -> 3.7
ce3466d204..3fc13c8688  3.6 -> 3.6
ecf8d07442..544fd7ea72  3.5 -> 3.5
